// ng-lightning/modals/modal.js
import { Component, Input, Output, ElementRef, Renderer, EventEmitter, HostListener, ViewChild, ContentChild } from '@angular/core';
import { toBoolean, uniqueId } from '../util/util';
import { NglModalFooter } from './footer';
import { NglModalHeaderTemplate } from './header';
export var NglModal = (function () {
    function NglModal(element, renderer) {
        this.element = element;
        this.renderer = renderer;
        this.header = '';
        this.headingId = uniqueId('modal_header');
        this.open = true;
        this.openChange = new EventEmitter();
        this._directional = false;
    }
    Object.defineProperty(NglModal.prototype, "directional", {
        get: function () {
            return this._directional;
        },
        set: function (directional) {
            this._directional = toBoolean(directional);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NglModal.prototype, "_open", {
        set: function (_open) {
            var _this = this;
            _open = toBoolean(_open);
            if (_open === this.open)
                return;
            if (_open) {
                setTimeout(function () { return _this.focusFirst(); });
            }
            this.open = _open;
        },
        enumerable: true,
        configurable: true
    });
    NglModal.prototype.close = function (evt) {
        if (evt) {
            evt.stopPropagation();
        }
        this.openChange.emit(false);
    };
    NglModal.prototype.stopPropagation = function (evt) {
        evt.stopPropagation();
    };
    NglModal.prototype.focusFirst = function () {
        this.renderer.invokeElementMethod(this.closeButton.nativeElement, 'focus', []);
    };
    NglModal.decorators = [
        { type: Component, args: [{
                    selector: 'ngl-modal',
                    template: "\n    <div [class.slds-fade-in-open]=\"open\" [ngClass]=\"[size ? 'slds-modal--' + size : '']\" [attr.aria-hidden]=\"!open\" role=\"dialog\" [attr.aria-labelledby]=\"headingId\" class=\"slds-modal\"><div class=\"slds-modal__container\"><div [class.slds-modal__header--empty]=\"!header &amp;&amp; !headerTpl\" class=\"slds-modal__header\"><button #closeButton type=\"button\" (click)=\"close()\" class=\"slds-button slds-button--icon-inverse slds-modal__close\"><svg nglIcon=\"close\" class=\"slds-button__icon slds-button__icon--large\"></svg><span class=\"slds-assistive-text\">Close</span></button><h2 *ngIf=\"header\" [id]=\"headingId\" class=\"slds-text-heading--medium\">{{header}}</h2><template *ngIf=\"headerTpl\" [ngTemplateOutlet]=\"headerTpl.templateRef\" [ngOutletContext]=\"{id: headingId}\"></template><ng-content select=\"[tagline]\"></ng-content></div><div class=\"slds-modal__content slds-p-around--medium\"><ng-content select=\"[body]\"></ng-content></div><div *ngIf=\"footer\" [class.slds-modal__footer--directional]=\"directional\" class=\"slds-modal__footer\"><template [ngTemplateOutlet]=\"footer.templateRef\"></template></div></div></div><div tabindex=\"0\" (focus)=\"focusFirst()\"></div><div [class.slds-backdrop--open]=\"open\" class=\"slds-backdrop\"></div>\n  ",
                    host: {
                        'tabindex': '0',
                    },
                },] },
    ];
    /** @nocollapse */
    NglModal.ctorParameters = [
        { type: ElementRef, },
        { type: Renderer, },
    ];
    NglModal.propDecorators = {
        'header': [{ type: Input },],
        'size': [{ type: Input },],
        'directional': [{ type: Input },],
        'closeButton': [{ type: ViewChild, args: ['closeButton',] },],
        '_open': [{ type: Input, args: ['open',] },],
        'openChange': [{ type: Output },],
        'headerTpl': [{ type: ContentChild, args: [NglModalHeaderTemplate,] },],
        'footer': [{ type: ContentChild, args: [NglModalFooter,] },],
        'close': [{ type: HostListener, args: ['keydown.esc', ['$event'],] },],
        'stopPropagation': [{ type: HostListener, args: ['click', ['$event'],] },],
    };
    return NglModal;
}());
;
